File: PDFTest/ViewController.m
//
//  ViewController.m
//  PDFTest
//
//  Created by Irena on 07.09.17.
//  Copyright © 2017 Irena. All rights reserved.
//

#import "ViewController.h"

@interface ViewController ()<UITextFieldDelegate>{
    // Responsible for display PDF file
    WebViewPDF *pdfViewer;
    __weak IBOutlet UITextField *txtSearch;

}

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    
    // Do any additional setup after loading the view.
    pdfViewer = [[WebViewPDF alloc] initWithFrame:self.view.bounds];
    CGRect frame = pdfViewer.frame;
    frame.origin.y += 60;
    pdfViewer.frame = frame;
    [self.view addSubview:pdfViewer];
    
    // Disbale scroll bounce
    pdfViewer.scrollView.bounces = NO;
}

-(void) viewDidAppear:(BOOL)animated
{
    [super viewDidAppear:animated];
    NSString *pdfFilePath = [[NSBundle mainBundle] pathForResource:@"compressed.tracemonkey-pldi-09" ofType:@"pdf"];
    [pdfViewer loadPDF:pdfFilePath];
    
    /*NSString *pdf = [[NSBundle mainBundle] pathForResource:@"compressed.tracemonkey-pldi-09" ofType:@"pdf"];
    NSURL* url = [NSURL URLWithString:pdf];
    NSURLRequest* request = [NSURLRequest requestWithURL:url];
    [pdfViewer loadRequest:request];*/
}


- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

#pragma mark - UITextfield delegates
-(BOOL) textFieldShouldReturn:(UITextField *)textField
{
    [pdfViewer search:textField.text];
    [textField resignFirstResponder];
    return YES;
}

-(BOOL)textField:(UITextField *)textField shouldChangeCharactersInRange:(NSRange)range replacementString:(NSString *)string
{
    NSString *finalString = [textField.text stringByReplacingCharactersInRange:range withString:string];
    
    {
        [pdfViewer search:finalString];
    }
    return YES;
}

#pragma mark - handel click event
- (IBAction)btnPreviusClicked:(id)sender {
    [pdfViewer previousSearchResult];
}

- (IBAction)btnNextClicked:(id)sender {
    [pdfViewer nextSearchResult];
}

//- (IBAction)btnHighlightedClicked:(UIButton*)sender {
//    [pdfViewer setHighliteSearch:sender.isSelected];
//}
//
//- (IBAction)btnCaseSensitiveClicked:(UIButton*)sender {
//    [pdfViewer setCase

@end

File: TestPDF/ViewController.swift
//
//  ViewController.swift
//  TestPDF
//
//  Created by Irena on 11.09.17.
//  Copyright © 2017 Irena. All rights reserved.
//

import UIKit

class ViewController: UIViewController {
    
    // Responsible for display PDF file
    var pdfViewer: WebViewPDF?

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        
        pdfViewer = WebViewPDF(frame: self.view.bounds)
        var frame: CGRect = pdfViewer!.frame
        frame.origin.y += 80
        pdfViewer?.frame = frame
        self.view.addSubview(pdfViewer!)
        // Disbale scroll bounce
        pdfViewer?.scrollView.bounces = false
        
        let filepath: String = Bundle.main.path(forResource: "compressed.tracemonkey-pldi-09", ofType: "pdf")!
        pdfViewer?.loadPDF(filePath: filepath)
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }
    
    // MARK: - UITextfield delegates
    func textFieldShouldReturn(_ textField: UITextField) -> Bool {
        pdfViewer?.search(textField.text!)
        textField.resignFirstResponder()
        return true
    }
    
    func textField(_ textField: UITextField, shouldChangeCharactersIn range: NSRange, replacementString string: String) -> Bool {
        let finalString: String = (textField.text! as NSString).replacingCharacters(in: range, with: string)
        do {
            pdfViewer?.search(finalString)
        }
        return true
    }
    
    // MARK: - handel click event
    @IBAction func btnPreviusClicked(_ sender: Any) {
        pdfViewer?.previousSearchResult()
    }
    
    @IBAction func btnNextClicked(_ sender: Any) {
        pdfViewer?.nextSearchResult()
    }
}

